modules/exploits/multi/acronis_cyber_protect_unauth_rce_cve_2022_3405.rb

```diff
@@ -0,0 +1,615 @@
+
+          The web console contains multiple possibilities to execute arbitrary commands on both the
+          agents (e.g., via PreCommands for a backup) and also the appliance (e.g., via a Validation
+          job on the agent of the appliance). These options can easily be set with the provided bearer
+          token, which leads to a complete compromise of all agents and the appliance itself.
+
+          You can either use the module `auxiliary/gather/acronis_cyber_protect_machine_info_disclosure`
+          to collect target info for exploitation in this module. Or just run this module standalone and
+          it will try to exploit the first online endpoint matching your target and payload settings
+          configured at the module.
+
+          Acronis Cyber Protect 15 (Windows, Linux) before build 29486 and
+          Acronis Cyber Backup 12.5 (Windows, Linux) before build 16545 are vulnerable.
+        },
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # Metasploit module
+          'Sandro Tolksdorf of usd AG.'             # discovery
+        ],
+        'References' => [
+          ['CVE', '2022-3405'],
+          ['URL', 'https://herolab.usd.de/security-advisories/usd-2022-0008/'],
+          ['URL', 'https://attackerkb.com/topics/WVI3r5eNIc/cve-2022-3405']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux', 'windows'],
+        'Privileged' => true,
+        'Arch' => [ARCH_CMD],
+        'Targets' => [
+          [
+            'Unix/Linux Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => ['windows'],
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2022-11-08',
+        'DefaultOptions' => {
+          'SSL' => true,
+          'RPORT' => 9877
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS],
+          'Reliability' => [REPEATABLE_SESSION]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'The URI of the vulnerable Acronis Cyber Protect/Backup instance', '/']),
+      OptString.new('HOSTID', [false, 'hostId value collected from recon module "auxiliary/gather/acronis_cyber_protect_machine_info_disclosure"', '']),
+      OptString.new('PARENTID', [false, 'parentId value collected from recon module "auxiliary/gather/acronis_cyber_protect_machine_info_disclosure"', '']),
+      OptString.new('KEY', [false, 'key value collected from recon module "auxiliary/gather/acronis_cyber_protect_machine_info_disclosure"', ''])
+    ])
+  end
+
+  # return first access_token or nil if not successful
+  def get_access_token1
```

Comment:
It looks like there's a lot of code that's been copy/pasted from the information disclosure module - is it worth adding a mixin for code share between the two modules :eyes:

Reply:
Yeah, agree. That would make sense.

Though, I do have a question.

How do get this new `mixin` shared between the two open PRs https://github.com/rapid7/metasploit-framework/pull/19582 and https://github.com/rapid7/metasploit-framework/pull/19583.
I am not sure how to do this.

Reply:
Hey @h00die-gr3y. You can just add the mixin to one of the PRs and I can make sure that one gets tested landed first, if you'd like.

Reply:
Okido, let me do that with PR https://github.com/rapid7/metasploit-framework/pull/19582

Reply:
Thank you!!